autorag/evaluate/metric/generation.py

```diff
@@ -144,6 +145,10 @@ def sem_score(generation_gt: List[str], pred: str, embedding_model: Optional[Bas
     gt_embeddings = embedding_model.get_text_embedding_batch(generation_gt)
     pred_embedding = embedding_model.get_text_embedding(pred)
 
+    del embedding_model
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
```

Comment:
Here is the issue.
As you see, this `sem_score` function will run so many times in `generation_metric` decorator. 
Then, the deletion of embedding_model at every `sem_score` function is super inefficient. Because of this, every row, the python must load all models from the start. It takes super long time for I/O.

[Suggest Solution]
Do not use generation_metric decorator at `sem_score`. Like meteor or rouge. 

Init embedding_model only one time. And delete it at the end, after all row evaluation is end.